integration/spark/src/main/scala/org/apache/carbondata/trash/DataTrashManager.scala

```diff
@@ -87,13 +108,70 @@ object DataTrashManager {
     }
   }
 
-  private def checkAndCleanTrashFolder(carbonTable: CarbonTable, isForceDelete: Boolean): Unit = {
+  /**
+   * Checks the size of the segment files as well as datafiles, this method is used before and after
+   * clean files operation to check how much space is actually freed, during the operation.
+   */
+  def getSizeScreenshot(carbonTable: CarbonTable): Long = {
+    val segmentPath = CarbonTablePath.getSegmentFilesLocation(carbonTable.getTablePath)
+    var size : Long = 0
+    if (!carbonTable.isHivePartitionTable) {
+      if (carbonTable.getTableInfo.getFactTable.getTableProperties.containsKey(
+          CarbonCommonConstants.FLAT_FOLDER)) {
+        // the size is table size + segment folder size - (metadata folder size + lockFiles size)
+        (FileFactory.getDirectorySize(carbonTable.getTablePath) + FileFactory.getDirectorySize(
```

Comment:
here do not calculate the fact folder size directly , because when there 1000s of segments it will be slow. We are already storing the data and index size in metadata, please make use of it and it will faster

Reply:
done